### PR TITLE
[HOPSWORKS-912] support image and avro types in featurestore

### PR DESCRIPTION
**Pull request by Limmen**
- Add avro spark dependency

## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/projects/HOPSWORKS/issues/HOPSWORKS-912

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the new behavior (if this is a feature change)?**
New datatypes supported for feature store training datasets (e.g Avro/Image)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by Limmen** [comment]
Tests: https://github.com/logicalclocks/hops-testing/pull/288

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java [hunk 1]
```diff
@@ -449,6 +459,10 @@ private boolean createConfigFiles(String confDirPath, String kernelsDir, String
       sparkMagicParams.put(Settings.SPARK_EXECUTOR_EXTRACLASSPATH, new ConfigProperty(
           "spark_executor_extraClassPath", HopsUtils.APPEND_PATH, extraClassPath));
 
+      sparkMagicParams.put(Settings.SPARK_PACKAGES, new ConfigProperty(
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java]: @Limmen @robzor92 @SirOibaf I suggest we add this avro dependency in spark-chef to the default spark jars. same way we do for parquet. If Hopsworks does not have internet access it won't be able to fetch the dependency from the maven central repo. If someone want to use different versions of these dependencies, they can add them as local resources or `spark.yarn.dist.jars`.

@Limmen Also, the property `spark_jars_packages` should not be ignore as users might provide additional dependencies.

Comment (Limmen) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java]: @tkakantousis It is not ignored. 
`
      //Prepare spark maven packages
      StringBuilder sparkPackages = new StringBuilder();
      //avro data source used in Feature Store
      sparkPackages.append(settings.getSparkAvroPackageName());
      //user-defined packages
      if(userSparkProperties.containsKey(Settings.SPARK_PACKAGES)){
        sparkPackages.append(",");
        sparkPackages.append(userSparkProperties.get(Settings.SPARK_PACKAGES));
      }
`
I've tested it on VM, you can add your own packages fine.

Comment (tkakantousis) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java]: @Limmen yes you are right, I mixed up the HopsUtils properties.

Comment (tkakantousis) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterConfigFilesGenerator.java]: @Limmen we don't need the localresources for this PR. LocalResources is meant for dependencies that are in HopsFs and the user wants to have them distributed to the executors. I think we can close this PR. We can the avro dependency in spark-chef ([link](https://github.com/logicalclocks/spark-chef/blob/3a962d079a2b600778e228616285a6a7a32cf095/recipes/install.rb#L74))together with the parquet dependencies. Then spark-avro will be in the classpath of the executors.

**Pull request merged**